content/200-concepts/100-components/02-prisma-client/050-filtering-and-sorting.mdx

```diff
@@ -0,0 +1,182 @@
+- [Filter _on_ related record properties](relation-queries#filtering) - for example, only return users _with_ at least one published post
+
+### Advanced field filtering
+
+#### `Json` field filtering
+
+#### Array / list field filtering
+
+### Case-insensitive filtering
+
+Case-insensitive filtering [is available as a feature for the PostgreSQL provider](case-sensitivity#options-for-case-insensitive-filtering). MySQL and MariaDB are case-insensitive by default, and do not require a Prisma Client feature to make case-insensitive filtering possible.
+
+To use case-insensitive filtering, add the `mode` property to a particular filter and specify `insensitive`:
+
+```ts
+const users = await prisma.user.findMany({
+  where: {
+    email: {
+      endsWith: "prisma.io",
+|      mode: "insensitive", // Default value: default
+    },
+    name: {
+       equals: "Archibald" // Default mode
+    }
+  },
+});
+```
+
+See also: [Case sensitivity](case-sensitivity)
+
+## Sorting
+
+Use [`orderBy`](../../../reference/api-reference/prisma-client-reference#orderby) <span class="api"></span> to sort a list of records or a nested list of records by a particular field or set of fields. For example, the following query returns all `User` records sorted by `role` and `name`, **and** each user's posts sorted by `title`:
+
+<CodeWithResult>
+
+<Cmd>
+
+```ts
+const usersWithPosts = await prisma.user.findMany({
+  orderBy: [
+    {
+      role: 'desc',
+    },
+    {
+      email: 'desc',
+    },
+  ],
+  include: {
+    posts: {
+      orderBy: {
+        title: "desc",
+      },
+      select: {
+        title: true,
+      },
+    },
+  },
+});
+```
+
+</Cmd>
+
+<CmdResult>
+
+```json
+[
+  {
+      "email": "kwame@prisma.io",
+      "id": 2,
+      "name": "Kwame",
+      "role": "USER",
+      "posts": [
+          {
+              "title": "Prisma in five minutes"
+          },
+          {
+              "title": "Happy Table Friends: Relations in Prisma"
+          }
+      ]
+  },
+  {
+      "email": "emily@prisma.io",
+      "id": 5,
+      "name": "Emily",
+      "role": "USER",
+      "posts": [
+          {
+              "title": "Prisma Day 2020"
+          },
+          {
+              "title": "My first day at Prisma"
+          },
+          {
+              "title": "All about databases"
+          }
+      ]
+  }
+]
+```
+
+</CmdResult>
+
+</CodeWithResult>
+
+> **Note**: You can also [sort lists of nested records when you use `findUnique`](#sort-one-users-nested-list-of-post-records)
+ to retrieve a single record by ID.
+
+### Sorting FAQs
+ 
+#### Can I sort by the property of a nested `model`?
+
+Follow [issue #249 on GitHub](https://github.com/prisma/prisma-client-js/issues/249).
+
+#### Can I sort by an aggregate value (for example, by user's post count)?
+
+Follow [issue #1 on GitHub](https://github.com/prisma/prisma-client-js/issues/1).
```

Comment:
seems like this was introduced with `groupBy`? https://github.com/prisma/prisma-client-js/issues/1